feat: Replace "we recommend" and "Codacy recommends" DOCS-497 (#1576)
codacy/docs: docs/repositories-configure/managing-branches.md

# Managing branches

{% if not extra.self_hosted %}

> Just like on your Git provider, every repository on Codacy has a default branch. Codacy always analyzes the default branch and loads its data first on dashboards. When you add a repository, the default branch on Codacy matches the default branch on your Git provider (typically `master` or `main`) and is the only one being analyzed.

Author: * I realized that here we're unclear on what a Codacy default branch is, vs. a Git provider default branch. I tried to clarify.
* Moreover, I found it hard to include the mention that Codacy analyzes multiple branches without sounding like it was just shoved in there. The whole paragraph is about default branches, and the next line introducing the steps should make it fairly clear that you can analyze other branches. I get that we may want to "push" the feature here a bit, but it still feels a bit off...

Author: Moreover:

* I included the bit about loading on dashboards, taken from the admonition.
* I removed the mention "consider matching the default on your provider to the default on Codacy." The default already does that, and I'm not sure there's much value. It's an edge case for which, hopefully, a user can notice what's going on independently. Just a guess though.

Author: This is a bigger edit and potentially out of scope for this task, but it just naturally came to light as I was editing. I'm open to moving this one to another task.

Reviewer: Maybe we're all overthinking this a bit :sweat_smile:, but I find that the new paragraph is more complex than the previous one.

I have a suggestion on how it could be simplified, but I agree that this larger edit deserves its own pull request. This way, we can already merge everything related to the "recommends" and we can fine-tune the "Managing branches" page separately.

> To change the default branch of your repository or start analyzing other branches:

Author: Quick reword, since we're at it

# Managing branches

{% if not extra.self_hosted %}

> Just like on your Git provider, every repository on Codacy has a default branch. Codacy always analyzes the default branch and loads its data first on dashboards. When you add a repository, the default branch on Codacy matches the default branch on your Git provider (typically `master` or `main`) and is the only one being analyzed.

Author: Yup, had the same feeling. I reverted leaving a minor edit: a mention that Codacy loads the default branch first (that was our whole point in the end). Let me know if it works like this.

```suggestion
Codacy automatically analyzes the default branch of your repository (typically `master` or `main` as configured on your Git provider) and loads its data first on dashboards. Codacy also supports analyzing multiple branches.
```

Author: @prcr
Let me know if this is better (it opened this new conversation for some reason)

Reviewer: Yes, it's more straight to the point now. :smile: Nice editing!

Reviewer: @nicklem I think you forgot to apply the changes that you were suggesting in this thread. :slightly_smiling_face:

Author: @prcr Good catch! Just opened https://github.com/codacy/docs/pull/1583 to fix

{% if not extra.self_hosted %}
Codacy automatically analyzes the default branch of your repository (typically `master` or `main` as configured on your Git provider), and also supports analyzing multiple branches.

!!! info

>     Codacy displays information for the default branch on dashboards. Consider matching the default branch on Codacy to the default branch on your Git provider.

Reviewer: Just two comments regarding this information:

- In the current version, users may understand that Codacy **only** displays information on dashboards for the default branch. We're not saying that Codacy displays those branches by default (implying that users can then change the branch). It becomes a bit tricky to add this information because of the repetition "By default, Codacy displays the default branch". :sweat_smile:
- I'd experiment with having this information directly as a paragraph (ditching the admonition). After all, it's the explanation of what a default branch is on the Codacy side.

Author: Yeah, it's a tricky sentence! I'll let it simmer until tomorrow.

Author: Following up on https://github.com/codacy/docs/pull/1576/commits/82bd26de66833ee8fa32b5ed3772f477ded3b23b

    Enabling a new branch triggers an initial analysis of that branch and the analysis results and information for that branch will become available after the analysis is complete.

> 1.  To change the default branch on Codacy, click the corresponding **Make default** link that appears when you hover the column **Default**. Changing the default branch on Codacy doesn't change the default branch on your Git provider.

Author: While we're clarifying what a default branch is, I realized that we should probably mention this. It's not obvious

Reviewer: Yes, good idea! :ok_hand:

1.  To change the default branch on Codacy, click the corresponding **Make default** link that appears when you hover the column **Default**.

>     Codacy preselects the default branch when loading dashboards. Consider matching the default branch on Codacy to the default branch on your Git provider.

Author: I reworded this one too for clarity

>     Codacy preselects the default branch when loading dashboards. Consider matching the default branch on Codacy to the default branch on your Git provider.

Reviewer: The "preselects" may be harder to understand. What about this "in-between" version?

```suggestion
    Codacy displays information for the default branch on dashboards. Consider matching the default branch on Codacy to the default branch on your Git provider.
```

>     Codacy displays information for the default branch on dashboards. Consider matching the default branch on Codacy to the default branch on your Git provider.

Reviewer 2: Isn't this recommendation a bit hidden down here? Does it make sense to move it to the top? (for example, inside an infobox after the first sentence)

Reviewer: Indeed, this should be part of the "concept" instead of the "task". :nerd_face:

Author: I moved the sentence to an infobox as suggested, although it reads a bit weird...

Reviewer 2: Looks good to me. 👍
